code/modules/wiremod/components/sensors/computervision.dm

```diff
@@ -0,0 +1,52 @@
+/**
+ * # Computer Vision Component
+ *
+ * Scans for mobs or objects
+ */
+/obj/item/circuit_component/computer_vision
+	display_name = "Computer Vision Component"
+	display_desc = "Scans its surroundings to find mobs or objects"
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+	var/datum/port/output/scanned_objects
+	var/scan_range = 7
```

Comment:
might want to lower this.

Reply:
the scan range? I did consider making it 5